Microsoft.Toolkit.Uwp.UI.Controls/Menu/MenuItem.cs

```diff
@@ -81,6 +81,13 @@ public MenuItem()
         {
             DefaultStyleKey = typeof(MenuItem);
             IsFocusEngagementEnabled = true;
+
+            IsEnabledChanged += (s, e) =>
```

Comment:
The registration for the event should be in method OnApplyTemplate and also we should unregister from the event as well :) for example you can do

`

/// <inheritdoc />
        protected override void OnApplyTemplate()
        {
            FlyoutButton = GetTemplateChild(FlyoutButtonName) as Button;
            _parentMenu = this.FindParent<Menu>();
            IsOpened = false;

            Items.VectorChanged -= Items_VectorChanged;
            IsEnabledChanged -= MenuItem_IsEnabledChanged;

            if (MenuFlyout == null)
            {
                MenuFlyout = new MenuFlyout();
            }
            else
            {
                MenuFlyout.Opened -= MenuFlyout_Opened;

                MenuFlyout.Closed -= MenuFlyout_Closed;
            }

            if (FlyoutButton != null)
            {
                FlyoutButton.PointerExited -= FlyoutButton_PointerExited;
                Items.VectorChanged += Items_VectorChanged;

                MenuFlyout.Opened += MenuFlyout_Opened;
                MenuFlyout.Closed += MenuFlyout_Closed;
                FlyoutButton.PointerExited += FlyoutButton_PointerExited;

                MenuFlyout.MenuFlyoutPresenterStyle = _parentMenu.MenuFlyoutStyle;
                ReAddItemsToFlyout();

                IsEnabledChanged += MenuItem_IsEnabledChanged;

                if (_isAccessKeySupported)
                {
                    FlyoutButton.AccessKey = AccessKey;
                    AccessKey = string.Empty;
                }
            }

            UpdateEnabledVisualState();

            base.OnApplyTemplate();
        }

        private void MenuItem_IsEnabledChanged(object sender, DependencyPropertyChangedEventArgs e)
        {
            var menuItemControl = (MenuItem)sender;
            menuItemControl.UpdateEnabledVisualState();
        }`